calico_kubernetes/calico_kubernetes.py

```diff
@@ -265,28 +265,58 @@ def _assign_container_ip(self):
 
         :return IPAddress which has been assigned
         """
+        def _assign(ip):
```

Comment:
Why have you made this a nested function, instead of a method in its own right? Reckon the call stack would be cleaner in the latter formulation.